src/service/resources-impl.js

```diff
@@ -487,6 +487,7 @@ export class Resources {
       resource.requestMeasure();
       dev().fine(TAG_, 'resource reused:', resource.debugid);
     } else {
+      this.relayoutAll_ = true;
```

Comment:
I think we should properly debounce it. Remeasuring everything in every cycle will be expensive. Also, by default adds do not run cycles at all. E.g. maybe something like this:
```
this.remeasuringPass_ = new Pass(() => {
  this.relayoutAll_ = true;
  this.schedulePass();
});

add(...) {
  ...
  this.remeasuringPass_.schedule(1000);
}
```
?